.azure-pipelines/generate-v1.0-sdk.yml

```diff
@@ -0,0 +1,172 @@
+      Write-Host "Path to typewriter.exe output $outputPath"
+      Write-Host "##vso[task.setvariable variable=OUTPUTPATH]$outputPath"
+
+      $cleanMetadata = "https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/clean_v10_metadata/cleanMetadataWithDescriptionsv1.0.xml"
+      Write-Host "Path to clean metadata $cleanMetadata"
+      Write-Host "##vso[task.setvariable variable=CLEANMETADATA]$cleanMetadata"
+
+      $branchName = "v1.0/pipelinebuild/$env:BUILD_BUILDID" # Match the spec in the GH Action
+      Write-Host "Branch path spec for the pull request will be $branchName"
+      Write-Host "##vso[task.setvariable variable=BRANCHNAME]$branchName"
+
+- task: PowerShell@2
+  displayName: 'Git: checkout dev'
+  inputs:
+    targetType: inline
+    pwsh: true
+    workingDirectory: '$(Build.SourcesDirectory)/msgraph-sdk-php'
+    script: |
+      git checkout dev | Write-Host
+
+- task: PowerShell@2
+  displayName: 'Git: branch from dev named with the build id: $(Build.BuildId)'
+  inputs:
+    targetType: inline
+    workingDirectory: '$(Build.SourcesDirectory)/msgraph-sdk-php'
+    script: |
+      Write-Host "The new branch name will be: $env:BRANCHNAME"
+      git checkout -B $env:BRANCHNAME | Write-Host
+
+- task: PowerShell@2
+  displayName: 'Git: set user config'
+  inputs:
+    targetType: inline
+    pwsh: true
+    workingDirectory: '$(Build.SourcesDirectory)/msgraph-sdk-php'
+    script: |
+      git config user.email "GraphTooling@service.microsoft.com"
+      git config user.name "Microsoft Graph DevX Tooling"
+
+- task: PowerShell@2
+  displayName: 'Remove generated files from the repo'
+  inputs:
+    targetType: inline
+    pwsh: true
+    script: |
+      Remove-Item -Recurse $env:REPOMODELSDIR | Write-Host
+      Write-Host "Removed the existing generated files in the repo." -ForegroundColor Green
+
+- task: PowerShell@2
+  displayName: 'Typewriter: generate v1.0 .NET files'
+  inputs:
+    targetType: filePath
+    pwsh: true
+    filePath: '$(Build.SourcesDirectory)/msgraph-metadata/scripts/runTypewriter.ps1'
+    arguments: '-verbosity Info -metadata $(CLEANMETADATA) -output $(OUTPUTPATH) -generationMode Files -l PHP'
+    workingDirectory: '$(Build.SourcesDirectory)' # Set the root for a multi-repo pipeline. /s
+  enabled: true
+
+- task: PowerShell@2
+  displayName: 'Copy generated files into the repo'
+  inputs:
+    targetType: inline
+    pwsh: true
+    script: |
+      $modelsDirectory = Join-Path $env:OUTPUTPATH "/com/microsoft/graph/model/"
+      Move-Item $modelsDirectory $env:REPOMODELSDIR
+      Write-Host "Moved the models from $modelsDirectory into the local repo." -ForegroundColor Green
+
+- task: PowerShell@2
+  displayName: 'Run PHP tests'
+  inputs:
+    targetType: inline
+    workingDirectory: '$(Build.SourcesDirectory)/msgraph-sdk-php'
+    pwsh: true 
+    script: |
+      $installDir = "c:\tools\php"
+      choco install composer --ia "/DEV=$installDir /PHP=$installDir"
```

Comment:
This choco install may no longer be necessary per this PR: https://github.com/actions/virtual-environments/pull/588